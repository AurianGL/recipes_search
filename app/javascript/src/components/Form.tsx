import * as React from "react";
import * as Yup from 'yup';
import { withFormik, FormikProps, FormikErrors, Form, Field, Formik } from 'formik';
import { useContext } from "react";
import { DispatchContext } from "../contexts/search_context";

export const MyForm = () => {
  const dispatch = useContext(DispatchContext);

  return (
    <Formik
      initialValues={{
        ingredient: '',
        quantity: 0,
        unity: 'quant'
      }}
      onSubmit={(values, { setSubmitting }) => {
        setSubmitting(true);
        dispatch({ type: 'ADD_INGREDIENT', payload: values });
        setSubmitting(false);
      }}
      validationSchema={Yup.object({
        ingredient: Yup.string()
          .required('Required'),
        quantity: Yup.number()
          .required('Required')
          .positive('Must be a positive number')
          .integer('Must be an integer'),
        unity: Yup.string()
          .required('Required')
          .oneOf(['weight', 'volume', 'quant'], 'Invalid unity')
      })}
    >
      {({ isSubmitting, touched, errors }) => (
        <Form className="grid gap-2 grid-cols-1 w-1/3 text-center">
          <h1>Search Ingredients</h1>
          <label htmlFor="ingredient">Ingredient</label>
          <Field name="ingredient" className="text-blue-900 rounded" />
          {touched.ingredient && errors.ingredient && <div className="text-red-600">{errors.ingredient}</div>}
          <label htmlFor="quantity">Quantity</label>
          <Field name="quantity" className="text-blue-900 rounded" />
          {touched.quantity && errors.quantity && <div className="text-red-600">{errors.quantity}</div>}
          <label htmlFor="unity">Measurement metric</label>
          <Field as="select" name="unity" className="text-blue-900 rounded">
            <option value="weight">Weight</option>
            <option value="volume">Volume</option>
            <option value="quant">Quant</option>
          </Field>
          {touched.unity && errors.unity && <div className="text-red-600">{errors.unity}</div>}


          <button type="submit" disabled={isSubmitting}>
            Submit
          </button>
        </Form>
      )}
    </Formik>
  );
}
