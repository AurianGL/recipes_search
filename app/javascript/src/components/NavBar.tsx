import * as React from "react";
import { Link } from "react-router-dom"

export const NavBar = () => {
  return (
    <nav>
      <ul className="grid gap-2 grid-cols-3 w-full  text-center p-4 bg-rose-900">
        <li >
          <Link className='bg-rose-600 hover:bg-rose-400 hover:text-black rounded px-3 py-2 cursor-pointer' to="/">Home</Link>
        </li>
        <li >
          <Link className='bg-rose-600 hover:bg-rose-400 hover:text-black rounded px-3 py-2 cursor-pointer' to={'/index'}>index</Link>
        </li>
        <li >
          <Link className='bg-rose-600 hover:bg-rose-400 hover:text-black rounded px-3 py-2 cursor-pointer' to={'/search'}>search</Link>
        </li>
      </ul>
    </nav>
  )
}
