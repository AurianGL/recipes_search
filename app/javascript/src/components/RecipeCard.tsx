import * as React from "react";
import { Link } from "react-router-dom";

export const RecipeCard = (props) => {
  const { recipe, withDetails } = props;
  return (
    <div className="bg-slate-400 rounded-lg p-3">
      <div className="font-bold">{recipe.title}</div>
      {withDetails && <><div>{recipe.ingredients.map((ingredient, index) => {
        return <div key={index}>{ingredient}</div>
      })}</div>
      <img src={recipe.image} alt="" className="max-h-36"/></>}
      {!withDetails && <Link to={`/recipes/${recipe.id}`}>Details</Link>}
    </div>
  )
}