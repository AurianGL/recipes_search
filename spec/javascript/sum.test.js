import React from 'react';
import renderer from 'react-test-renderer';

import {MyForm} from 'src/components/Form';

test('rendered component', () => {
  const component = renderer.create(<MyForm/>)
    
  expect(component.toJSON()).toMatchSnapshot();
});