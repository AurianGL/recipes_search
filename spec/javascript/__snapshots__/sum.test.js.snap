// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered component 1`] = `
<form
  action="#"
  className="grid gap-2 grid-cols-1 w-1/3 text-center"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <h1>
    Search Ingredients
  </h1>
  <label
    htmlFor="ingredient"
  >
    Ingredient
  </label>
  <input
    className="text-blue-900 rounded"
    name="ingredient"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="quantity"
  >
    Quantity
  </label>
  <input
    className="text-blue-900 rounded"
    name="quantity"
    onBlur={[Function]}
    onChange={[Function]}
    value={0}
  />
  <label
    htmlFor="unity"
  >
    Measurement metric
  </label>
  <select
    className="text-blue-900 rounded"
    name="unity"
    onBlur={[Function]}
    onChange={[Function]}
    value="quant"
  >
    <option
      value="weight"
    >
      Weight
    </option>
    <option
      value="volume"
    >
      Volume
    </option>
    <option
      value="quant"
    >
      Quant
    </option>
  </select>
  <button
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
